bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.openhab.binding.renault.internal.renault.api.Car;
+import org.openhab.binding.renault.internal.renault.api.MyRenaultHttpSession;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RenaultHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class RenaultHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RenaultHandler.class);
+
+    private RenaultConfiguration config = new RenaultConfiguration();
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private HttpClient httpClient;
+
+    public RenaultHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // This binding only polls status data automatically.
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(RenaultConfiguration.class);
+
+        // Validate configuration
+        if (this.config.myRenaultUsername.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Username is empty!");
+            return;
+        }
+        if (this.config.myRenaultPassword.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Password is empty!");
+            return;
+        }
+        if (this.config.locale.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Location is empty!");
+            return;
+        }
+        if (this.config.vin.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "VIN is empty!");
+            return;
+        }
+        if (this.config.refreshInterval < 1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The refresh interval mush to be larger than 1");
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Background initialization:
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
+        }
```

Comment:
Here is how to avoid compilation warnings (NPE):
```
        ScheduledFuture<?> job = pollingJob;
        if (job == null || job.isCancelled()) {
            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
        }
```

Reply:
@lolodomo Thanks for the tip but I am confused how this works?  You create a new reference to a nullable object then check the ref is not null.  How is this different to checking the original ref to the nullable object?  Happy to adopt this technique but I would like to understand it.

Reply:
As pollingJob is a class member, the compiler assumes that it could be updated at any time by any other thread.
Using a local variable solves this problem.

Reply:
@lolodomo Merci! I wrote a unit test to test my understanding of the "=" char.  Turns out I got it wrong.  I thought it was a pointer to the memory of the object and adding a new var to the the old pointer was just a new ref to the memory location but your comment and my unit test show that it copies the value too!    ( I have only been programming Java for 25 years ! )  Now I understand this properly this so thanks.